[bookmarks] switch spatial bookmarks dialog to dockwidget with toolbar

slarosa opened the pull request:
![bookmarks_dock](https://cloud.githubusercontent.com/assets/1374682/8160322/e9936276-136e-11e5-9c43-0098b5124059.png)

slarosa committed:
[bookmarks] switch spatial bookmarks dialog to dockwidget with toolbar

NathanW2 commented:
Oh I like that nice job.  Can we commit this during a feature freeze?

On Mon, Jun 15, 2015 at 11:04 PM, Salvatore Larosa <notifications@github.com
> wrote:

> [image: bookmarks_dock]
> <https://cloud.githubusercontent.com/assets/1374682/8160322/e9936276-136e-11e5-9c43-0098b5124059.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/qgis/QGIS/pull/2142
> Commit Summary
>
>    - [bookmarks] switch spatial bookmarks dialog to dockwidget with
>    toolbar
>
> File Changes
>
>    - *M* images/images.qrc
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-0> (3)
>    - *A* images/themes/default/mActionSharing.svg
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-1> (398)
>    - *A* images/themes/default/mActionSharingExport.svg
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-2> (421)
>    - *A* images/themes/default/mActionSharingImport.svg
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-3> (421)
>    - *M* src/app/qgisapp.cpp
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-4> (93)
>    - *M* src/app/qgisapp.h
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-5> (12)
>    - *M* src/app/qgsbookmarks.cpp
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-6> (61)
>    - *M* src/app/qgsbookmarks.h
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-7> (19)
>    - *M* src/ui/qgsbookmarksbase.ui
>    <https://github.com/qgis/QGIS/pull/2142/files#diff-8> (156)
>
> Patch Links:
>
>    - https://github.com/qgis/QGIS/pull/2142.patch
>    - https://github.com/qgis/QGIS/pull/2142.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2142>.
>

nyalldawson commented:
@slarosa what about making the help button part of the toolbar too? That's taking up a lot of vertical space which could be better used by the list widget

slarosa commented:
@nyalldawson yes I will do, but now I trying to understand why it is failing on travis. On my machine the PyQgsAppStartup is passing.

slarosa commented:
@nyalldawson done in 4469c7b.

![schermata del 2015-06-15 18 28 41](https://cloud.githubusercontent.com/assets/1374682/8164898/7bb7f9c4-138c-11e5-8e6c-71277ea2618d.png)

nyalldawson commented:
@slarosa looks much better, thank you! The startup test should be disabled on Travis since a week or so ago - is your branch up to date?

slarosa commented:
@nyalldawson if I am not wrong the startup test is failing on testPyQgisStartupEnvVar, @m-kuhn has disabled the testPluginPath since ed47ef9, do you mean that?

btw my branch is updated to 7a3a857

m-kuhn commented:
@slarosa the testPluginPath was disabled since it acted strange on a fix by @NathanW2 and the failing test could not be reproduced locally. I hope to be able to re-enable it once a more solid fix is implemented.

Does the failing startupEnvVar test pass for you if you run it locally?

slarosa commented:
@m-kuhn yes, the PyQgsAppStartup test is fine for me if I run it locally but:
```
The following tests FAILED:
	  1 - qgis_qgistest (Failed)
	 37 - qgis_composerutils (Failed)
	 44 - qgis_composermapgridtest (Failed)
	 53 - qgis_composertablev2test (Failed)
	 58 - qgis_composerscalebartest (Failed)
	 69 - qgis_maprotationtest (Failed)
	 72 - qgis_legendrenderertest (Failed)
	 87 - qgis_scalecombobox (Failed)
	102 - PyQgsLocalServer (Failed)
	134 - PyQgsPalLabelingCanvas (Failed)
	135 - PyQgsPalLabelingComposer (Failed)
	136 - PyQgsPalLabelingServer (Failed)
	146 - PyQgsSpatialiteProvider (Failed)
```

nyalldawson commented:
@slarosa what platform is that?

slarosa commented:
@nyalldawson debian wheezy.

m-kuhn commented on src/app/qgisapp.cpp:
```diff
@@ -603,6 +603,10 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, QWidget * parent,
 
+  // Bookmarks dock
+  mBookMarksDockWidget = new QgsBookmarks( this );
+  mBookMarksDockWidget->setObjectName( "BooknarksDockWidget" );
```
That should probably not be Booknarks ;)

slarosa replied:
@m-kuhn :smile_cat:

slarosa commented:
@nyalldawson @m-kuhn one more detailed report here: http://dash.orfeo-toolbox.org/viewTest.php?onlyfailed&buildid=186968

m-kuhn commented:
Server:
> Could not locate server binaries
> Find 'spawn-fcgi': 
> Find 'lighttpd': 

Spatialite:

> ```
> cannot start a transaction within a transaction
> AddGeometryColumn() error: unexpected metadata layout
> E
> ======================================================================
> ERROR: setUpClass (__main__.TestQgsSpatialiteProvider)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/sam/pacchetti_gis/QGIS/tests/src/python/test_provider_spatialite.py", line 70, in setUpClass
>     cur.execute(sql)
> OperationalError: table test_pg has no column named geometry
> ```
Not sure where that comes from... Maybe spatialite version?

That one may be because of the locale?
http://dash.orfeo-toolbox.org/testDetails.php?test=30943502&build=186968

Then we have some font issues (damn you, rendering tests with fonts!)... @nyalldawson mask?

nyalldawson commented:
@m-kuhn yep, its on my to-do list to update the labeling test masks... Painful stuff!

m-kuhn commented:
@slarosa I just incresed the timeout on master. Can you rebase on top of it and see if it that helps?

slarosa commented:
@m-kuhn that was what I was thinking to do :)

nirvn commented:
@slarosa I love it. It'd be nice to have this for 2.10, fits well with other UI improvements recently committed.

m-kuhn commented:
Travis now produces debug output so it should be possible to see up to which place the appstartup test runs.

m-kuhn commented:
So it seems the failure is related to bookmarks. At least messages concerning bookmarks appear as last breath.
However, I am not quite sure how to interpret this.

```
Warning: QSqlDatabase: QSQLITE driver not loaded

Stacktrace (piped through c++filt):

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c48cf]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c4589]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(qt_message_output(QtMsgType, char const*)+0x2e)[0x7f58748b941e]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(+0x718bf)[0x7f58748b98bf]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(qWarning(char const*, ...)+0x94)[0x7f58748ba044]

/usr/lib/x86_64-linux-gnu/libQtSql.so.4(+0x151c4)[0x7f58799051c4]

/usr/lib/x86_64-linux-gnu/libQtSql.so.4(QSqlDatabase::addDatabase(QString const&, QString const&)+0x19)[0x7f58799055c9]

/home/travis/build/qgis/QGIS/build/output/bin/qgis(QgsBookmarks::QgsBookmarks(QWidget*)+0x3d8)[0x57fb78]

/home/travis/build/qgis/QGIS/build/output/bin/qgis(QgisApp::QgisApp(QSplashScreen*, bool, QWidget*, QFlags<Qt::WindowType>)+0xb49)[0x4cd509]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c7ad8]

/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xed)[0x7f587180176d]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c40d5]

src/core/qgsmessagelog.cpp: 45: (logMessage) [79ms] 2015-06-16T16:22:23 Qt[1] QSqlDatabase: QSQLITE driver not loaded

Warning: QSqlDatabase: available drivers: QMYSQL3 QMYSQL

Stacktrace (piped through c++filt):

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c48cf]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c4589]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(qt_message_output(QtMsgType, char const*)+0x2e)[0x7f58748b941e]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(+0x718bf)[0x7f58748b98bf]

/usr/lib/x86_64-linux-gnu/libQtCore.so.4(qWarning(char const*, ...)+0x94)[0x7f58748ba044]

/usr/lib/x86_64-linux-gnu/libQtSql.so.4(+0x1525c)[0x7f587990525c]

/usr/lib/x86_64-linux-gnu/libQtSql.so.4(QSqlDatabase::addDatabase(QString const&, QString const&)+0x19)[0x7f58799055c9]

/home/travis/build/qgis/QGIS/build/output/bin/qgis(QgsBookmarks::QgsBookmarks(QWidget*)+0x3d8)[0x57fb78]

/home/travis/build/qgis/QGIS/build/output/bin/qgis(QgisApp::QgisApp(QSplashScreen*, bool, QWidget*, QFlags<Qt::WindowType>)+0xb49)[0x4cd509]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c7ad8]

/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xed)[0x7f587180176d]

/home/travis/build/qgis/QGIS/build/output/bin/qgis[0x4c40d5]

src/core/qgsmessagelog.cpp: 45: (logMessage) [6ms] 2015-06-16T16:22:23 Qt[1] QSqlDatabase: available drivers: QMYSQL3 QMYSQL
```

m-kuhn commented:
Maybe add this to apt-get install in .travis.yml: libqt4-sql-sqlite

slarosa commented:
> Maybe add this to apt-get install in .travis.yml: libqt4-sql-sqlite

@m-kuhn is that reliable to be done? Do you think we can try? or it makes no sense...

m-kuhn commented:
@slarosa absolutely, if that fixes it it was just a missing dependency on the CI infrastructure. And if we decide that this dependency is mandatory for QGIS to run we can document it and add it to packaging if required and that's it.

slarosa commented:
I added the share icon (as from #2146) and rebased the branch. @m-kuhn are you ok to add the new dependency package on the CI infrastructure? otherwise this cannot be merged.

m-kuhn commented:
Absolutely positive with the dependency, no worries!!

slarosa commented:
manually merged in f79d747
